dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -12,39 +12,27 @@ def update_last_confirmation_date
 
   # PATCH /api/v1/users_school_infos/<id>/school_info
   def update
+    existing_school_info = current_user.school_info
     if school_info_params[:school_id].present?
-      existing_school_info = current_user.school_info
-      new_school = SchoolInfo.find_or_create_by(school_info_params)
-      if existing_school_info && !existing_school_info.complete?
-        current_user.update!(school_info_id: new_school.id)
+      new_school = SchoolInfo.find_or_create_by!(school_info_params.merge(validation_type: SchoolInfo::VALIDATION_NONE))
+      if existing_school_info == new_school && existing_school_info.complete?
```

Comment:
Shouldn't we update the last confirmation date whether or not the previous school info was complete or not?